yjit/src/codegen.rs

```diff
@@ -3979,6 +3963,42 @@ fn gen_send_cfunc(
     asm.cmp(CFP, stack_limit);
     asm.jbe(counted_exit!(ocb, side_exit, send_se_cf_overflow).into());
 
+
+    if flags & VM_CALL_ARGS_SPLAT != 0 {
+        if cfunc_argc > 0 {
+            argc = push_splat_args(argc, ctx, asm, ocb, side_exit, cfunc_argc as u32);
+        } else {
+            // This is a variadic c function and we'd need to pop
+            // each of the elements off the array, but the array may be dynamically sized
+            gen_counter_incr!(asm, send_args_splat_variadic);
+            return CantCompile
+        }
+    }
+
+    // pop_args_splat does a ctx.stack_pop so we can no longer side exit
+    let side_exit: Option<()> = None;
```

Comment:
It looks like `side_exit` is no longer used after this point, so we could probably remove this?

Reply:
The point of this is to be a guard for the future. If people start using side_exit after this, they will now get an error at compile time rather than just undefined behavior. I talked to @noahgibbs about this and he was down with the idea. But happy to remove if we'd rather not.

Reply:
I would prefer to remove it for now. Ideally the guards and side-exits should happen earlier.

Reply:
Cool, I got rid of it